File: altigo_web/src/core/api/apiClient.ts
import axios from 'axios';
import { env } from '../config/env';

const TOKEN_STORAGE_KEY = 'altigo_auth_token';

export const tokenStorage = {
  get: () => localStorage.getItem(TOKEN_STORAGE_KEY),
  set: (token: string) => localStorage.setItem(TOKEN_STORAGE_KEY, token),
  clear: () => localStorage.removeItem(TOKEN_STORAGE_KEY),
};

export const apiClient = axios.create({ baseURL: env.apiBaseUrl });

apiClient.interceptors.request.use((config) => {
  const token = tokenStorage.get();
  if (token) {
    config.headers = config.headers ?? {};
    config.headers.Authorization = `Bearer ${token}`;
  }
  return config;
});

File: altigo_app/lib/core/network/api_client.dart
import 'package:dio/dio.dart';
import '../config/env.dart';
import '../services/secure_storage_service.dart';

/// Thin wrapper around Dio that injects the JWT (when present) into every
/// outgoing request and centralizes the base URL configuration.
class ApiClient {
  ApiClient._internal() {
    _dio = Dio(BaseOptions(baseUrl: Env.apiBaseUrl, connectTimeout: const Duration(seconds: 15)));

    _dio.interceptors.add(
      InterceptorsWrapper(
        onRequest: (options, handler) async {
          final token = await SecureStorageService.instance.readToken();
          if (token != null) {
            options.headers['Authorization'] = 'Bearer $token';
          }
          handler.next(options);
        },
      ),
    );
  }

  static final ApiClient instance = ApiClient._internal();
  late final Dio _dio;

  Dio get dio => _dio;
}
